app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/MapsActivity.java

```diff
@@ -92,6 +112,116 @@ public void onSearchTextChanged(String oldQuery, String newQuery) {
+                                            ", " + MapsActivity.this.lastLocation.getLongitude(),
+                                    dest.toString(), dest.getName());
+                        } else {
+                            FetchPlaceRequest request = FetchPlaceRequest.builder(dest.toString(),
+                                    Arrays.asList(
+                                            com.google.android.libraries.places.api.model.Place.Field.LAT_LNG)).build();
+
+                            placesClient.fetchPlace(request).addOnSuccessListener((response) -> {
+                                launchRoute(MapsActivity.this.lastLocation.getLatitude() +
+                                                ", " + MapsActivity.this.lastLocation.getLongitude(),
+                                        response.getPlace().getLatLng().latitude + ", "
+                                                + response.getPlace().getLatLng().longitude,
+                                        dest.getName());
+
+                            });
+                        }
+                    }
+                }).start();
+            }
+
+            @Override
+            public void onSearchAction(String currentQuery) {
+            }
+        });
+    }
+
+    private void launchRoute(String start, String end, String name) {
+        Map<String, String> mapString = new HashMap<String, String>();
+        mapString.put("start", start);
+        mapString.put("end", end);
+        mapString.put("arriveBy", String.valueOf(false));
+        mapString.put("destinationName", name);
+
+        Calendar calendar = Calendar.getInstance(
+                TimeZone.getTimeZone("\"America/NewYork\""));
+        int secondsEpoch = (int) (calendar.getTimeInMillis() / 1000L);
+        mapString.put("time", String.valueOf(secondsEpoch));
+
+        Endpoint searchEndpoint = new Endpoint()
+                .queryItems(mapString)
+                .path("route")
+                .method(Endpoint.Method.GET);
+
+        FutureNovaRequest.make(
+                Route[].class, searchEndpoint).thenAccept(
+                (APIResponse<Route[]> response) -> {
+
+                    ArrayList<Route> routes = new ArrayList<Route>();
+                    for (Route r : response.getData()) {
+                        routes.add(r);
+                    }
+                    runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            if (routes.size() > 0) {
+                                getController().drawRoutes(routes.get(0), routes);
+                                getSearchView().setSearchText(name);
+                                getSearchView().clearSearchFocus();
+                            } else {
+                                Toast.makeText(MapsActivity.this,
+                                        "Something went wrong, we can't provide a route.",
```

Comment:
is this copy the one from the design? just double checking (the copy as in the text on this line) –– there are some cases where we don't have any routes

Reply:
I couldn't really find this in the design, so I just wrote up a quick line to toss in the a Toast. Toasts don't look that great though, so I'll change that at a later date to look a bit nicer.